src/torrent/data/block.cc

```diff
@@ -97,18 +97,18 @@ Block::insert(PeerInfo* peerInfo) {
 
   m_notStalled++;
 
-  auto itr = m_queued.insert(m_queued.end(), new BlockTransfer());
-
-  (*itr)->set_peer_info(peerInfo);
-  (*itr)->set_block(this);
-  (*itr)->set_piece(m_piece);
-  (*itr)->set_state(BlockTransfer::STATE_QUEUED);
-  (*itr)->set_request_time(this_thread::cached_seconds().count());
-  (*itr)->set_position(0);
-  (*itr)->set_stall(0);
-  (*itr)->set_failed_index(BlockFailed::invalid_index);
-
-  return (*itr);
+  auto& block = m_queued.emplace_back(new BlockTransfer);
```

Comment:
Write this as `auto*` to be more explicit about code intentions.

Reply:
Hmmm using non ref auto is a copy. I guess that’s fine for normal pointers.

Reply:
so this is interesting:

```
git grep auto\\\*
src/protocol/peer_chunks.h:  auto*               download_cache()              { return &m_downloadCache; }
src/protocol/peer_chunks.h:  auto*               upload_queue()                { return &m_uploadQueue; }
src/protocol/peer_chunks.h:  const auto*         upload_queue() const          { return &m_uploadQueue; }
src/protocol/peer_chunks.h:  auto*               cancel_queue()                { return &m_cancelQueue; }
```

it's more common to use non pointer auto it seems.
```

Reply:
OTOH it’s clear this is a pointer from the right hand side. Which is how auto works anyway.

Reply:
auto& is more appropriate IMO. Let’s say this gets converted to a smart pointer. & is needed to avoid copies.

Reply:
Made the change.